python/src/genetic_search.py

```diff
@@ -96,6 +98,8 @@ class GeneticCandidateGenerator(CandidateGenerator[Netlist]):
 
     _rng: np.random.Generator
 
+    _get_initial_population: Callable[["GeneticCandidateGenerator"], Sequence[Netlist]]
```

Comment:
imo it would be cleaner to just have `_initial_population_override: Optional[Sequence[Netlist]]`, set it in the static `create` method, then in `get_initial_population` check do a 
```
if self._initial_population_override is not None`:
    return self._initial_population_override
```

If you decide not to do that for whatever reason I think this `_get_initial_population` has to be wrapped in Optional like the callable are in the search algorithm class?

Reply:
I think this would also get rid of the mypy error

Reply:
Yeah that seems reasonable, python with typing does seem to make overriding methods annoying. I am too lazy to rewrite it that way now though so I will take the one line fix of how I decided to do it.